mutation/mutation_compactor.hh

```diff
@@ -293,6 +317,7 @@ private:
             _max_purgeable = _get_max_purgeable(*_dk, is_shadowable);
         }
         auto ret = t.timestamp < _max_purgeable.timestamp;
+        timestamp_source = _max_purgeable.source;
```

Comment:
Please avoid output parameters, they reduce readability at the call site. Return a struct.

Reply:
Also, perhaps we can drop the optional by adding an enumeration option for none.

Reply:
Done